### ISSUE TITLE
twisted/mail/maildir.py's importing of stuff from cred is messy

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4475]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4475|
|Type|defect|
|Created|2010-05-31 22:42:51Z|

`twisted/mail/maildir.py` does this:

```
from twisted import cred
import twisted.cred.portal
import twisted.cred.credentials
import twisted.cred.checkers
import twisted.cred.error
```

This confuses pyflakes, and probably some human readers too, as the expression `twisted.cred.portal` (nor any of the others) does not appear in this file.  Instead, `cred.portal` (etc) are used.

It would be better to just import the modules, or even the attributes of the modules being used.

Attachments:

* [remove_unnecessary_imports.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/65e/65e32ffe0a235f88e218a889fee50ee4f7cc1e4b/b379cf2608b8fca9669a1086691a9dd3d3ce4c96.patch) (535 bytes) - added by igorsobreira on 2010-09-28 03:20:55Z - 
* [modifying_imports.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/65e/65e32ffe0a235f88e218a889fee50ee4f7cc1e4b/2c7630183cc222e233b2d5a93a94f413943b757f.patch) (1982 bytes) - added by igorsobreira on 2010-09-30 02:20:54Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4475 4475
type__defect defect
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__easy easy
time__1275345771000000 1275345771000000
changetime__1286075529000000 1286075529000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to igorsobreira|
|-|-|

Thanks for the patch.  This is about half of the change I would like to see.  `twisted/mail/maildir.py` still uses names like `cred.portal.IRealm` and `cred.credentials.IUsernamePassword`.  Instead it should use `portal.IRealm` or just `IRealm`.  The code only works with this patch applied because `maildir.py` imports `twisted.mail.mail` and `twisted.mail.mail` imports `twisted.cred.portal` and so on for the other cred modules.

**Comment by twisted-trac** [comment]
|<img alt="igorsobreira's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|igorsobreira commented|
|-|-|

Thanks for the feedback. I've attached a new patch with different imports.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [30084]) Apply modifying_imports.patch 

refs [#4475](https://github.com/twisted/twisted/issues/4475)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  This looks just right now, applying.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [30086]) Merge mail-imports-4475

Author: igorsobreira
Reviewer: exarkun
Fixes: [#4475](https://github.com/twisted/twisted/issues/4475)

Change the unusual import style in twisted/mail/maildir.py to something
more common.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
